.github/actions/install-languages-cli/action.yml

```
@@ -0,0 +1,34 @@
+name: Install Languages CLI
+description: Dowloads the Languages CLI from a known release or using Cargo
+
+inputs:
+  download_url:
+    required: false
+    description: The url to download the CLI binary from
+    default: ""
+  update_rust_toolchain:
+    required: false
+    description: Should we run `rustup update` as part of this composite action?
+    default: "true"
+
+runs:
+  using: composite
+  steps:
+    - name: Update Rust toolchain
+      shell: bash
+      if: inputs.update_rust_toolchain == 'true' && inputs.download_url == ''
+      run: rustup update
+
+    - name: Build actions binary
+      shell: bash
+      if: inputs.download_url == ''
+      run: cargo install --git https://github.com/heroku/languages-github-actions.git
+
+    - name: Download actions binary
+      shell: bash
+      if: inputs.download_url != ''
+      run: |
+        mkdir -p ${{ runner.tool_cache }}/languages-github-actions
+        curl ${{ inputs.download_url }} --output ${{ runner.temp }}/actions.tar.gz -L --retry 5 --retry-all-errors
```

(The `--fail` ensures curl exits non-zero rather than the script trying to run `tar` anyway, and the timeout options prevent hangs)

```suggestion
        curl ${{ inputs.download_url }} --output ${{ runner.temp }}/actions.tar.gz -L --fail --retry 5 --retry-all-errors --connect-timeout 10 --max-time 60
```

(Also I've checked that it's ok to use `` since it was only added in 7.71.0, but it's fine since the GHA 22.04 runner [has v7.81.0-1ubuntu1.13](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#installed-apt-packages))